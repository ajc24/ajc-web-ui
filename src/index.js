/* Export all of the header components */
export { default as Header } from './ui-elements/header/Header';

/* Export all of the icon components */
export { default as SmallDownArrowIcon } from './ui-elements/icons/SmallDownArrowIcon';
export { default as SmallRightArrowIcon } from './ui-elements/icons/SmallRightArrowIcon';
