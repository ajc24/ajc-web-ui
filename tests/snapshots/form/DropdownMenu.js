import 'jsdom-global/register';
import React from 'react';
import { TestDev } from 'ajc-jest-enzyme';
import { DropdownMenu } from '../../../src';

describe('DropdownMenu', () => {
  const mockOptionsList = [
    {
      textContent: 'One',
      value: 'one',
    },
    {
      textContent: 'Two',
      value: 'two',
    },
    {
      textContent: 'Three',
      value: 'three',
    },
    {
      textContent: 'Four',
      value: 'four',
    },
    {
      textContent: 'Five',
      value: 'five',
    },
  ];

  describe('Default props and rendering - Standalone component, no additional spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DropdownMenu id="dropdown-menu-id" labelText="Test Dropdown Menu Label Text:" optionsList={mockOptionsList} />
        </React.Fragment>
      );
    });
    
    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Standalone component with additional spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DropdownMenu id="dropdown-menu-id" labelText="Test Dropdown Menu Label Text:" optionsList={mockOptionsList} additionalUpperSpacing={true} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });

  describe('Transferred props and rendering - Dialog component, no additional spacing', () => {
    let jsonSnapshot;

    beforeAll(() => {
      jsonSnapshot = TestDev.createSnapshot(
        <React.Fragment>
          <DropdownMenu id="dropdown-menu-id" labelText="Test Dropdown Menu Label Text:" optionsList={mockOptionsList} isDialogFormItem={true} />
        </React.Fragment>
      );
    });

    it('verifies the snapshot for the component', () => {
      expect(jsonSnapshot).toMatchSnapshot();
    });
  });
});
