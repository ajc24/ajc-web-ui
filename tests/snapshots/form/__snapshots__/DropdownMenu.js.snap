// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownMenu Default props and rendering - Standalone component, no additional spacing verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-spacing-bottom-small ajc-text-black ajc-font-default"
>
  <label
    className="ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-standalone"
    htmlFor="dropdown-menu-id"
  >
    Test Dropdown Menu Label Text:
  </label>
  <select
    className="ajc-select ajc-font-default"
    defaultValue="one"
    id="dropdown-menu-id"
    name="dropdown-menu-id"
  >
    <option
      title="One"
      value="one"
    >
      One
    </option>
    <option
      title="Two"
      value="two"
    >
      Two
    </option>
    <option
      title="Three"
      value="three"
    >
      Three
    </option>
    <option
      title="Four"
      value="four"
    >
      Four
    </option>
    <option
      title="Five"
      value="five"
    >
      Five
    </option>
  </select>
</div>
`;

exports[`DropdownMenu Transferred props and rendering - Dialog component, no additional spacing verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-spacing-bottom-small ajc-text-black ajc-font-default"
>
  <label
    className="ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-dialog"
    htmlFor="dropdown-menu-id"
  >
    Test Dropdown Menu Label Text:
  </label>
  <select
    className="ajc-select ajc-font-default"
    defaultValue="one"
    id="dropdown-menu-id"
    name="dropdown-menu-id"
  >
    <option
      title="One"
      value="one"
    >
      One
    </option>
    <option
      title="Two"
      value="two"
    >
      Two
    </option>
    <option
      title="Three"
      value="three"
    >
      Three
    </option>
    <option
      title="Four"
      value="four"
    >
      Four
    </option>
    <option
      title="Five"
      value="five"
    >
      Five
    </option>
  </select>
</div>
`;

exports[`DropdownMenu Transferred props and rendering - Standalone component with additional spacing verifies the snapshot for the component 1`] = `
<div
  className="ajc-flex-column ajc-flex-justify-left ajc-background-transparent ajc-spacing-bottom-small ajc-text-black ajc-font-default"
>
  <label
    className="ajc-row ajc-flex-justify-left ajc-spacing-bottom-small ajc-spacing-bottom-reduced ajc-label-standalone ajc-spacing-top-large"
    htmlFor="dropdown-menu-id"
  >
    Test Dropdown Menu Label Text:
  </label>
  <select
    className="ajc-select ajc-font-default"
    defaultValue="one"
    id="dropdown-menu-id"
    name="dropdown-menu-id"
  >
    <option
      title="One"
      value="one"
    >
      One
    </option>
    <option
      title="Two"
      value="two"
    >
      Two
    </option>
    <option
      title="Three"
      value="three"
    >
      Three
    </option>
    <option
      title="Four"
      value="four"
    >
      Four
    </option>
    <option
      title="Five"
      value="five"
    >
      Five
    </option>
  </select>
</div>
`;
